src/compiler/scala/tools/nsc/javac/JavaParsers.scala

```diff
@@ -808,6 +827,61 @@ trait JavaParsers extends ast.parser.ParsersCommon with JavaScanners {
       })
     }
 
+    def recordDecl(mods: Modifiers): List[Tree] = {
+      accept(RECORD)
+      val pos = in.currentPos
+      val name = identForType()
+      val tparams = typeParams()
+      val header = formalParams()
+      val superclass = javaLangRecord()
+      val interfaces = interfacesOpt()
+      val (statics, body) = typeBody(RECORD, name)
+
+      // Records generate a canonical constructor and accessors, unless they are manually specified
+      var generateCanonicalCtor = true
+      var generateAccessors = header
+        .view
+        .map { case ValDef(_, name, tpt, _) => name -> tpt }
+        .toMap
+      for (DefDef(_, name, List(), List(params), tpt, _) <- body) {
+        if (name == nme.CONSTRUCTOR && params.size == header.size) {
+          val ctorParamsAreCanonical = params.lazyZip(header).forall {
+            case (ValDef(_, _, tpt1, _), ValDef(_, _, tpt2, _)) => tpt1 equalsStructure tpt2
+            case _ => false
+          }
+          if (ctorParamsAreCanonical) generateCanonicalCtor = false
+        } else if (generateAccessors.contains(name) && params.isEmpty) {
+          generateAccessors -= name
+        }
+      }
+
+      // Generate canonical constructor and accessors, if not already manually specified
+      val accessors = generateAccessors
+        .map { case (name, tpt) =>
+          DefDef(Modifiers(Flags.JAVA), name, List(), List(), tpt, blankExpr)
```

Comment:
Should this have `Flags.CASEACCESSOR`, `Flags.PARAMACCESSOR`, or `Flags.SYNTHETIC`?

Reply:
I think just `JAVA` is good